### PR TITLE
build: add source maps into NPM distribution

### PR DESCRIPTION
**Pull request by jasonaden**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -115,8 +115,9 @@ downlevelES2015() {
       cp ${file} ${ts_file}
 
       echo "======           $TSC ${ts_file} --target es5 --module es2015 --noLib"
-      ($TSC ${ts_file} --target es5 --module es2015 --noLib) > /dev/null 2>&1 || true
-      rm -f ${ts_file}
+      ($TSC ${ts_file} --target es5 --module es2015 --noLib --sourceMap) > /dev/null 2>&1 || true
+      mapSources "${BASH_REMATCH[1]}${2:-".es5.js"}"
+#      rm -f ${ts_file}
```
Comment (IgorMinar) [inline build.sh]: remove?

File: build.sh [hunk 2]
```diff
@@ -288,6 +296,33 @@ addNgcPackageJson() {
...
+# Arguments:
+#   param1 - JavaScript file on which to re-map sources
+# Returns:
+#   None
+#######################################
+mapSources() {
+  if [[ -f "${1}.map" ]]; then
+    $MAP_SOURCES -f "${1}"
+  fi
+#  # Iterate over the files in this directory, looking for .map files
```
Comment (IgorMinar) [inline build.sh]: remove the commented out code?

File: karma-js.conf.js [hunk 3]
```diff
@@ -41,7 +41,7 @@ module.exports = function(config) {
       'node_modules/reflect-metadata/Reflect.js',
       'tools/build/file2modulename.js',
       'test-main.js',
-      {pattern: 'dist/all/empty.*', included: false, watched: false},
+      {pattern: 'dist/all/@angular/empty.*', included: false, watched: false},
```
Comment (IgorMinar) [inline karma-js.conf.js]: is this right?

**Review by IgorMinar — COMMENTED** [review]
This looks very promising! Can you please remove the dead code?

File: build.sh (hunk 1 above)
Comment (jasonaden) [inline build.sh]: Done

File: build.sh (hunk 2 above)
Comment (jasonaden) [inline build.sh]: Done

File: karma-js.conf.js (hunk 3 above)
Comment (jasonaden) [inline karma-js.conf.js]: This is part of another PR #15073 (which also needs review). I needed this so I could run the tests locally.

**Pull request merged**